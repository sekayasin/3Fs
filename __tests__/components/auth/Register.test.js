import React from 'react';
import { shallow } from 'enzyme';
import {
  Register,
  mapStateToProps
} from '../../../src/components/auth/Register';

const registerUser = jest.fn();
const props = {
  onChange: jest.fn(),
  onSubmit: jest.fn(),
  registerUser
};
const state = {
  auth: '',
  errors: 'sekabira'
};

describe('test user registration', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Register {...props} />);
  });
  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('receives props', () => {
    wrapper.setProps({ errors: 'error' });
    expect(wrapper.state('errors')).toEqual('error');
  });
  it('test onChange', () => {
    wrapper
      .instance()
      .onChange({ target: { name: 'password', value: 'hajat❤️' } });
    expect(wrapper.state('password')).toEqual('hajat❤️');
  });
  it('test onSubmit', () => {
    wrapper.instance().onSubmit({ preventDefault: jest.fn() });
    expect(registerUser).toBeCalled();
  });
  it('test mapStateToProps', () => {
    expect(mapStateToProps(state).errors).toEqual('sekabira');
  });
});
