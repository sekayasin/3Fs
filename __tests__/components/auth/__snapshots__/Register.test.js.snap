// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test user registration matches snapshot 1`] = `ShallowWrapper {}`;
